### PR TITLE
[SYM-4799] Update Segment Log Destination example to use runtime-connector module 2.x

### PR DESCRIPTION
**Pull request by Arianna2028**
## Description
- Updates the `basic/segment_log_destination` example to use version 2.x of the [runtime-connector](https://registry.terraform.io/modules/symopsio/runtime-connector/aws/latest) module.
- Updates the usages of `main` to reference the `local.environment_name` variable instead, for consistency.
- Updates the Slack channel referenced in `impl.py` to match the standard recommendation of `#sym-requests`.
- Updates the Sym Terraform Provider to v3 and the `sym_flow.implementation` to be file contents, which is required in v3.
- Removes `runtime_id` from the `sym_environment`, since it is not needed to use Segment.

## Testing
Tested using the alpha version of the runtime-connector module and the following `override.tf`:
```hcl
provider "sym" {
  org = "sym"
}

module "runtime_connector" {
  source  = "symopsio/runtime-connector/aws"
  version = "2.0.0-alpha"

  environment_name = local.environment_name
  sym_account_id = "906191524257"
}

```

### CODE REVIEW

**Comment by Arianna2028** [comment]
Current dependencies on/for this PR:
* main
  * **PR #68** <a href="https://app.graphite.dev/github/pr/symopsio/examples/68" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/symopsio/examples/68?utm_source=stack-comment).

File: basic/segment_log_destination/segment_log_destination.tf [hunk 1]
```diff
@@ -1,10 +1,10 @@
-# aws secretsmanager put-secret-value --secret-id "main/segment-write-key" --secret-string "YOUR-SEGMENT-WRITE-KEY"
+# aws secretsmanager put-secret-value --secret-id "${local.environment_name}/segment-write-key" --secret-string "YOUR-SEGMENT-WRITE-KEY"
 resource "aws_secretsmanager_secret" "segment_write_key" {
-  name        = "main/segment-write-key"
+  name        = "${local.environment_name}/segment-write-key"
```
Comment (llam15) [inline basic/segment_log_destination/segment_log_destination.tf:3]:
```suggestion
  name        = "sym/${local.environment_name}/segment-write-key"
```

File: basic/segment_log_destination/secrets.tf [hunk 2]
```diff
@@ -23,7 +23,7 @@ EOT
 
 resource "aws_iam_role_policy_attachment" "attach_secrets_manager_access" {
```
Comment (llam15) [inline basic/segment_log_destination/secrets.tf:24]: Let's use the secrets manager module here

File: basic/segment_log_destination/segment_log_destination.tf [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
   description = "Segment Write Key for Sym Audit Logs"
 
   tags = {
-    # This SymEnv tag is required and MUST match the SymEnv tag in the 
+    # This SymEnv tag is required and MUST match the SymEnv tag in the
     # aws_iam_policy.secrets_manager_access in your `secrets.tf` file
```
Comment (llam15) [inline basic/segment_log_destination/segment_log_destination.tf:8]: After updating `secrets.tf`, let's update this comment

**Review by llam15 — APPROVED** [review]

**Pull request merged**